app/routes/datasets/versions.py

```diff
@@ -240,13 +240,36 @@ async def append_to_version(
     # For the background task, we only need the new source uri from the request
     input_data = {"creation_options": deepcopy(default_asset.creation_options)}
     input_data["creation_options"]["source_uri"] = request.source_uri
+
+    # If source_driver is "text", this is a datapump request
+    if input_data["creation_options"]["source_driver"] != "text":
+        # Verify that source_driver is not None
+        if input_data["creation_options"]["source_driver"] is None:
```

Comment:
The pydanic model for appending should be updated to take in `creation_options`, right? That'd be the right place to validate input like [so](https://github.com/wri/gfw-data-api/blob/master/app/models/pydantic/creation_options.py#L186):